### DISCUSSION TITLE
How to modify an existing wikibase/wikidata item?

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [LeMyst/WikibaseIntegrator#328] [Q&A] [answered]
Is there an example for modifying a few properties of an existing item somewhere?

**Comment by LeMyst** [comment]
Hello @WolfgangFahl ,

If you use v0.12.x, you can check my recent project wd-population:
https://github.com/LeMyst/wd-population/blob/5c394c7942e54ddeda2e10b9d5704fe9832899fc/departements.py#L100

Between line 100 and 101, you have and update of ranks for all P1082 claims.

At line 104, you have the insertion of a new claim P1082, of type Quantity, property P1082, with references (line 49-53) and qualifiers (line 44-47) and the rank.

**Reply by JeongYunLee** [reply]
Can I get some more simple examples? I'm using v0.12

**Reply by LeMyst** [reply]
Hello @JeongYunLee 

Can you be more specific of the type of examples you want? Explain your issue and what you try to accomplish.
It's better if you create your own Q&A topic : https://github.com/LeMyst/WikibaseIntegrator/discussions/new/choose

Best Regards,

Myst

**Reply by LeMyst** [reply]
You can also found some examples in the notebooks folder of the repository:
https://github.com/LeMyst/WikibaseIntegrator/tree/master/notebooks

**Comment by WolfgangFahl** [comment]
@LeMyst Thank you for your swift response:

_If you use v0.12.x, _ ... no i don't since it's not been released on pypi.

The key line seems to 

be https://github.com/LeMyst/wd-population/blob/5c394c7942e54ddeda2e10b9d5704fe9832899fc/departements.py#L85
```python
wb_item = ItemEntity(api=wbi).get(id_item)
```
where the existing item is fetched.

**Comment by LeMyst** [comment, marked as answer]
If you use v0.11, you can check the /test/ folder in the "master" branch.
Like in https://github.com/LeMyst/WikibaseIntegrator/blob/master/test/test_wbi_core.py (line 36 per exemple)

```python
instance_of_append = wbi_datatype.ItemID(prop_nr='P31', value='Q1234', if_exists='APPEND')
item = wbi_core.ItemEngine(item_id="Q2", data=[instance_of_append])
item.write()
```

I didn't use v0.11 since months ago, I maybe forgot some parameters
